src/sdk/namespace_fs.js

```diff
@@ -2578,10 +2578,19 @@ class NamespaceFS {
         }
     }
 
+    /**
+     * _delete_path_dirs deletes all the paths in the hierarchy that are empty after a successful delete
+     * if dir is not empty it will stop at the first non empty dir
+     * if dir is also a directory object (disabled mode optimization - CONTENT_DIR xattr is on the directory) we stop the loop
+     * @param {String} file_path 
+     * @param {nb.NativeFSContext} fs_context 
+     */
     async _delete_path_dirs(file_path, fs_context) {
         try {
             let dir = path.dirname(file_path);
             while (dir !== this.bucket_path) {
+                const dir_stat = await nb_native().fs.stat(fs_context, dir);
+                if (dir_stat.xattr && dir_stat.xattr[XATTR_DIR_CONTENT]) break;
```

Comment:
I think that adding the explicit compare to undefined can help reading, as it is might be confusing if were not focus that the dir_stat.xattr[XATTR_DIR_CONTENT] of empty directory object is '0' (string with 0).
```suggestion
                if (dir_stat.xattr && dir_stat.xattr[XATTR_DIR_CONTENT] !== undefined) break;
```